ts/components/screens/BaseScreenComponent.tsx

```diff
@@ -57,7 +57,11 @@ interface OwnProps {
   headerBody?: React.ReactNode;
   headerBackgroundColor?: ColorValue;
   appLogo?: boolean;
-  isSearchAvailable?: boolean;
+  isSearchAvailable?: {
+    enabled: true;
+    searchType?: SearchType;
+    onSearchTap?: () => void;
+  };
```

Comment:
```suggestion
  isSearchAvailable?: ComponentProps<typeof BaseHeader>["isSearchAvailable"];
```
In this way we can define this prop once a time in one place